File: test-suite-java/src/test/java/micronaut/example/OpenSearchTest.java
package micronaut.example;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import micronaut.example.service.Movie;
import micronaut.example.service.MovieService;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;
import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
class OpenSearchTest {
    @Test
    void testOpenSearch(MovieService movieService) {
        String imdb = "KJFDOD";
        String title = "Die Hard";
        movieService.saveMovie(new Movie("KJFDOD", title));
        await().atMost(10, SECONDS).until(() ->
            movieService.searchMovies(title) != null
        );
        Movie result = movieService.searchMovies(title);
        Assertions.assertNotNull(result);
        assertEquals(title, result.getTitle());
        assertEquals(imdb, result.getImdb());
    }
}

File: test-suite-python/src/test/python/micronaut/example/OpenSearchTest.py
import time
from typing import Annotated

from jakarta.inject import Inject
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from micronaut.example.service.Movie import Movie
from micronaut.example.service.MovieService import MovieService


# The OpenSearch container address is supplied by micronaut.example.support.OpenSearchTestConfigurer for the
# "opensearch" environment
@MicronautTest(environments=["opensearch"])
class OpenSearchTest:
    movie_service: Annotated[MovieService, Inject]

    @Test
    def test_open_search(self) -> None:
        imdb = "KJFDOD"
        title = "Die Hard"
        self.movie_service.save_movie(Movie(imdb, title))

        deadline = time.time() + 10
        while self.movie_service.search_movies(title) is None and time.time() < deadline:
            time.sleep(0.5)

        result = self.movie_service.search_movies(title)
        assert result is not None
        assert result.title == title
        assert result.imdb == imdb
